.github/workflows/ci.yml

```diff
@@ -31,77 +30,27 @@ env:
   CI_FLEET_PROVISIONING_ROLE: ${{ secrets.AWS_CI_FLEET_PROVISIONING_ROLE }}
   CI_DEVICE_ADVISOR: ${{ secrets.AWS_CI_DEVICE_ADVISOR_ROLE }}
   CI_MQTT5_ROLE: ${{ secrets.AWS_CI_MQTT5_ROLE }}
+  CI_BUILD_AND_TEST_ROLE: arn:aws:iam::180635532705:role/V2_SDK_Unit_Testing
```

Comment:
Minor: I'd prefer if this was hidden in a secret. Technically there is no reason to put it in a secret, as it cannot be used without authenticating the IAM role on a per-repo basis, but I feel some additional security through obscurity couldn't hurt. It also makes it easier to change the IAM role in the future without needing to make a commit/PR.

Reply:
I think we should aim for no secrets and yes that requires quite a bit more work.  I also think we should move tests away from requiring the test fixture to query secrets and instead make everything environmental injection via script from builder actions.  That gives us a pattern that is the same across all repos, regardless of language, and the amount of resources and set up is significantly lessened.

Reply:
Okay. As a first step, I'll work on removing the IAM role secrets first 👍